WooCommerce/Classes/ViewModels/Order Details/OrderStatusListViewModel.swift

```diff
@@ -0,0 +1,67 @@
+import Foundation
+import Yosemite
+import class AutomatticTracks.CrashLogging
+
+final class OrderStatusListViewModel {
+    private var status: OrderStatusEnum?
+    private var dataSource: OrderStatusListDataSource
+
+    init(status: OrderStatusEnum, dataSource: OrderStatusListDataSource) {
+        self.status = status
+        self.dataSource = dataSource
+    }
+
+    func configureResultsController(tableView: UITableView) {
+        dataSource.startForwardingEvents(to: tableView)
+        do {
+            try dataSource.performFetch()
+        } catch {
+            CrashLogging.logError(error)
+        }
+        tableView.reloadData()
+    }
+
+    /// Return the number of statuses for the view.
+    ///
+    func statusCount() -> Int {
+        dataSource.statusCount()
+    }
+
+    /// Return the index of the current order status.
+    ///
+    func indexOfCurrentOrderStatus() -> IndexPath? {
+        guard let foundStatus = dataSource.statuses().filter({ $0.status == status }).first else {
```

Comment:
super nit: You don't need the `(` and `)` in the filter closure
```suggestion
        guard let foundStatus = dataSource.statuses().filter{ $0.status == status }.first else {
```

Reply:
I'm going to keep it. Xcode warns: "Trailing closure in this context is confusable with the body of the statement; pass as a parenthesized argument to silence this warning"